src/tracked_unique.rs

```diff
@@ -0,0 +1,152 @@
+//! A small change tracker to wrap around a unique so you can determine if the value changes between updates.
+use tracing::trace_span;
+
+use crate::prelude::*;
+
+use core::any::type_name;
+use std::{fmt, ops::Deref, ops::DerefMut};
+
+pub struct TrackedValue<T: 'static>(InnerTrackedState, T);
+
+pub struct TrackedMut<'a, T: 'static>(UniqueViewMut<'a, TrackedValue<T>>);
+pub struct Tracked<'a, T: 'static>(UniqueView<'a, TrackedValue<T>>);
+
+impl<'a, T: 'static + Send + Sync> Borrow<'a> for Tracked<'a, T> {
+    fn borrow(
+        all_storages: &'a AllStorages,
+        all_borrow: Option<SharedBorrow<'a>>,
+    ) -> Result<Self, error::GetStorage>
+    where
+        Self: Sized,
+    {
+        Ok(Tracked(Borrow::borrow(all_storages, all_borrow)?))
+    }
+}
+
+unsafe impl<'a, T: 'static + Send + Sync> BorrowInfo for Tracked<'a, T> {
+    fn borrow_info(mut info: &mut Vec<info::TypeInfo>) {
+        UniqueView::<'a, T>::borrow_info(&mut info);
+    }
+}
+
+impl<'a, T: 'static + Send + Sync> Borrow<'a> for TrackedMut<'a, T> {
+    fn borrow(
+        all_storages: &'a AllStorages,
+        all_borrow: Option<SharedBorrow<'a>>,
+    ) -> Result<Self, error::GetStorage>
+    where
+        Self: Sized,
+    {
+        Ok(TrackedMut(Borrow::borrow(all_storages, all_borrow)?))
+    }
+}
+
+unsafe impl<'a, T: 'static + Send + Sync> BorrowInfo for TrackedMut<'a, T> {
+    fn borrow_info(mut info: &mut Vec<info::TypeInfo>) {
+        UniqueViewMut::<'a, T>::borrow_info(&mut info);
+    }
+}
+
+#[derive(PartialEq)]
+enum InnerTrackedState {
+    New,
+    Modified,
+    NoChanges,
```

Comment:
```suggestion
    Unchanged,
```

Reply:
Any precedent for this rename or just preference?